### ISSUE TITLE
Is there a way to set an SBO term for a function definition?

### ISSUE THREAD

**Issue by draeger** [sys-bio/tellurium#340]
```
function relActFun(a)
  a + 2;
end

model m
  relActFun.sboTerm = SBO:0000487;

  // Rate Rules:
  a' = 7*relActFun(time);

  // Variable initializations:
  a = 5;

  // Other declarations:
  var a;
end
```

While other objects can be annotated using a syntax such as `a.sboTerm = SBO:XXXX;`, the line `relActFun.sboTerm = SBO:0000487;` causes an error. If I take it out, it works. So how to annotate a function in Antimony?

**Comment by 0u812** [comment]
Hi Andreas! I hadn’t thought of annotating functions. I’ll try to add that feature to the next release.

**Comment by draeger** [comment]
That would be nice! Thanks.

**Comment by hsauro** [comment]
Kyle, how are the annotations handled at the moment, is this a new antimony
syntax?

Herbert

**Comment by 0u812** [comment]
@hsauro at the moment they're handled by Tellurium Python code. I'll add documentation.

**Comment by luciansmith** [comment]
For what it's worth, I would annotate the function in the function declaration, not in a model:

    function relActFun(a)
      sboTerm = SBO:0000487;
      a + 2;
    end

You could annotate models that way, too.

This means that you couldn't create a model with a parameter called 'sboTerm', but perhaps we don't need to worry about that?

**Comment by draeger** [comment]
Excellent, thanks!

**Comment by 0u812** [comment]
My pleasure!

**Comment by luciansmith** [comment]
Yup; the other SBO-setting syntaxes are similarly a post-processing step
from Tellurium directly.  Shouldn't be overly difficult to add it to
Antimony directly; added it as
https://sourceforge.net/p/antimony/bug-reports/48/

-Lucian

**Comment by 0u812** [comment]
Hi Andreas, the problem turned out to be that the parser I wrote wasn't handling comments correctly when they occurred on the same line as an SBO term. I fixed the issue and uploaded version 2.0.18 (Mac and Linux binaries uploaded, still working on Windows). Please let me know if you still have trouble with it. I tested it with the following code:

```
function relActFun(a)
  sboTerm = SBO:0000487; # can be commented out & example still works
  a + 2;
end

model m
  relActFun.sboTerm = SBO:0000487; # can be commented out & example still works

  // Rate Rules:
  a' = 7*relActFun(time);

  // Variable initializations:
  a = 5;

  // Other declarations:
  var a;
end
```

Next cell:
```python
print(m.getSBML())
```
Returns:
```xml
...
<functionDefinition sboTerm="SBO:0000487" id="relActFun">
...
```

**Comment by draeger** [comment]
Excellent work, thanks! It works perfectly now!
